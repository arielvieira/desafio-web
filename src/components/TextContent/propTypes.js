import PropTypes from 'prop-types';

const TextContentPropTypes = {
  body: PropTypes.string,
  createAt: PropTypes.string,
  isRepo: PropTypes.bool,
  title: PropTypes.string,
  url: PropTypes.string,
};

export default TextContentPropTypes;
