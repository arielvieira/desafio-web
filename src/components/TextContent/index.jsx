import React from 'react';
import { NavLink } from 'react-router-dom';

// propTypes
import propTypes from './propTypes';

// Utils
import { stringUtils, dateUtils } from '../../utils';

const TextContent = (props) => {
  const {
    body,
    title,
    url,
    createAt,
    isRepo,
  } = props;

  const nav = isRepo
    ? (
      <NavLink className="list-item__header--link" to={url}>
        {title}
      </NavLink>
    ) : (
      <a className="list-item__header--link" href={url}>
        {title}
      </a>
    );

  const bodyText = stringUtils.limitString(body, 75);
  const dateText = createAt && dateUtils.toLocaleDateString(createAt);

  return (
    <div className="list-item__box">
      <div className="list-item__header">
        <h2 className="list-item__header--title">
          {nav}
        </h2>
      </div>
      <div className="list-item__body">
        <p className="list-item__body--text">
          {bodyText}
        </p>
        {createAt && (
          <p>
            Date:
            {dateText}
          </p>
        )}
      </div>
    </div>
  );
};

TextContent.defaultProps = {
  body: '',
  createAt: '',
};

TextContent.propTypes = {
  body: propTypes.body,
  createAt: propTypes.createAt,
  isRepo: propTypes.isRepo.isRequired,
  title: propTypes.title.isRequired,
  url: propTypes.url.isRequired,
};

export default TextContent;
