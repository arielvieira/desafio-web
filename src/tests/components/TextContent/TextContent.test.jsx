/* eslint-disable */
import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TextContent from '../../../components/TextContent';

enzyme.configure({ adapter: new Adapter() });

describe('TextContent', () => {
  it('should render TextContent correctly', () => {
    const wrapper = shallow(
      <TextContent
        title="postcss"
        body="Transforming styles with JS plugins"
        url="pullrequests/postcss/postcss"
        isRepo
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
