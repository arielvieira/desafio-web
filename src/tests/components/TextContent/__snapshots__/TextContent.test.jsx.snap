// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContent should render TextContent correctly 1`] = `
<div
  className="list-item__box"
>
  <div
    className="list-item__header"
  >
    <h2
      className="list-item__header--title"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="list-item__header--link"
        to="pullrequests/postcss/postcss"
      >
        postcss
      </NavLink>
    </h2>
  </div>
  <div
    className="list-item__body"
  >
    <p
      className="list-item__body--text"
    >
      Transforming styles with JS plugins
    </p>
  </div>
</div>
`;
